Добавить MinIO (#35)

birthdaysgift opened the pull request:
Так как мы планируем хранить аватарки пользователей и желаний в MinIO, необходимо добавить базовую конфигурацию минио для разработки, тестов и CI.

В рамках этой задачи необходимо:

- найтроить minio для работы в трех различных окружениях - `envs/local/dev`, `envs/local/test` и `envs/ci` для каждой конфигурации необходимо добавить `docker-compose.yml` с описанием контейнеров и `.env` файл с описанием необходимых переменных окружения

- добавить параметры соединения с минио в `src/config.py`

- добавить класс `Minio` в `src/shared/minio.py`, который будет отвечать за взаимодействие с минио

- добавить создание файла в минио в роут `create_entity`

- добавить тест для проверки состояния минио для роута `create_entity`

Rey-Lane reviewed (approved):
Approve. Единственное, что мне бросилось в глаза - это `MINIO_HOST`. Почему-то в случае с базой данных хост и порт мы разделяем на две константы, а в случае хранилища нет. Если так и задумывалось, то ок, если случайно так получилось и ты согласен разделить - раздели)

birthdaysgift commented on src/shared/types.py:
```diff
@@ -0,0 +1,12 @@
+from src.shared.enum import Enum
+
+
+class UrlSchema(Enum):
+    """Enum class for URL schemas."""
+
+    HTTP = "HTTP"
+    HTTPS = "HTTPS"
```
Новый энам для юрл схем

birthdaysgift commented on src/config.py:
```diff
@@ -13,6 +14,12 @@
 class Config(BaseSettings):
     """Application config."""
 
+    MINIO_HOST: str
+    MINIO_PORT: str
+    MINIO_ROOT_PASSWORD: str
+    MINIO_ROOT_USER: str
+    MINIO_SCHEMA: UrlSchema
```
Разделил хост, порт и схему

birthdaysgift commented on src/shared/minio.py:
```diff
@@ -0,0 +1,71 @@
+        :param args: other args for minio.Minio class
+        :param kwargs: other kwargs for minio.Minio class
+        """
+        super().__init__(f"{host}:{port}", user, password, *args, secure=(schema is UrlSchema.HTTPS), **kwargs)
```
Этот конструктор подгоняет наши параметры под параметры конструктора `minio.Minio`.

И волки целы и овцы сыты.

birthdaysgift reviewed (commented):
Изначально сделал `MINIO_HOST`, который содержит и хост и порт, потому что именно в таком виде конструктор `minio.Minio` принимает параметр соединения. А ещё он вместо схемы (http или https) почему-то принимает `secure`, который `True` или `False`. Их конструктор довольно странно смотрится и поэтому мы его переопределяем в своём классе `Minio`.

Так что да, поскольку мы всё равно можем сгладить различия между нашими переменными окржения и стандартным конструктором `Minio`, то почему бы и не разделить переменные и сделать по человечески.

По итогу теперь у нас в `MINIO_HOST` передаётся нормальный хост, в `MINIO_PORT` соотвественно порт, и в `MINIO_SCHEMA` будет лежать юрл схема.

birthdaysgift committed:
Добавить MinIO (#35)

Так как мы планируем хранить аватарки пользователей и желаний в MinIO, необходимо добавить базовую конфигурацию минио для разработки, тестов и CI.

В рамках этой задачи необходимо:

- найтроить minio для работы в трех различных окружениях - `envs/local/dev`, `envs/local/test` и `envs/ci` для каждой конфигурации необходимо добавить `docker-compose.yml` с описанием контейнеров и `.env` файл с описанием необходимых переменных окружения

- добавить параметры соединения с минио в `src/config.py`

- добавить класс `Minio` в `src/shared/minio.py`, который будет отвечать за взаимодействие с минио

- добавить создание файла в минио в роут `create_entity`

- добавить тест для проверки состояния минио для роута `create_entity`
